src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -421,37 +456,42 @@ protected long fetchOffsetTokenWithRetry() {
   }
 
   /**
-   * Fallback function to execute when retries for fetching getOffsetToken for channel fails.
+   * If a valid offset is found from snowflake, we will reset the topicPartition with
+   * (offsetReturnedFromSnowflake + 1).
+   *
+   * <p>Please note, this is done only after a failure in fetching offset. Idea behind resetting
+   * offset (1 more than what we found in snowflake) is that Kafka should send offsets from this
+   * offset number so as to not miss any data.
    *
-   * <p>Fallback function is to re-open the channel and fetch the latestOffsetToken one more time
-   * after reopen is successful.
+   * @return offset which was last present in Snowflake
+   */
+  private long offsetTokenFallbackSupplier() {
+    final long offsetRecoveredFromSnowflake = getRecoveredOffsetFromSnowflake();
+    final long offsetToResetInKafka = offsetRecoveredFromSnowflake + 1L;
+    sinkTaskContext.offset(topicPartition, offsetToResetInKafka);
```

Comment:
any benefits on NOT resetting the offset if both sides are equal already?

Reply:
We actually would not know through sinktaskcontext on what it feels is the last committed offset or what is the committed offset at consumer level. So it is probably just better to reset it no matter what consumer thinks.